src/serlio/PRTContext.cpp

```diff
@@ -75,19 +75,11 @@ PRTContext::PRTContext(const std::vector<std::wstring>& addExtDirs) : mPluginRoo
 	}
 	else {
         theCache.reset(prt::CacheObject::create(prt::CacheObject::CACHE_TYPE_DEFAULT));
-        mResolveMapCache = new ResolveMapCache(prtu::getProcessTempDir(SRL_TMP_PREFIX));
+        mResolveMapCache = std::make_unique<ResolveMapCache>(prtu::getProcessTempDir(SRL_TMP_PREFIX));
     }
 }
 
 PRTContext::~PRTContext() {
-	if (mResolveMapCache) { // TODO: smart ptr
-		delete mResolveMapCache;
-		mResolveMapCache = nullptr;
-	}
-
-	theCache.reset();
-	thePRT.reset();
-
```

Comment:
i think PRT currently requires to destroy the caches before shutting down PRT (need to verify this separately, deep rabbit hole).
as destruction order depends on member order (inverse), you are lucky ;)
maybe its best to leave the resets in, to write it explicitely.

Reply:
oops, good to know! reinstated the resets and added a comment